src/StoreKeeper/WooCommerce/B2C/Backoffice/Pages/Tabs/ExportTab.php

```diff
@@ -14,37 +16,50 @@
-use StoreKeeper\WooCommerce\B2C\Tools\IniHelper;
 use StoreKeeper\WooCommerce\B2C\Tools\Language;
-use Throwable;
 
 class ExportTab extends AbstractTab
 {
     use FormElementTrait;
 
-    const ACTION_EXPORT = 'export-action';
     const ACTION_GENERATE_SKU_FROM_TITLE = 'generate-sku-from-title';
 
     public function __construct(string $title, string $slug = '')
     {
         parent::__construct($title, $slug);
-
-        $this->addAction(self::ACTION_EXPORT, [$this, 'exportAction']);
         $this->addAction(self::ACTION_GENERATE_SKU_FROM_TITLE, [$this, 'generateSkuFromTitleAction']);
     }
 
-    protected function exportAction()
+    public function register(): void
     {
-        if (array_key_exists('type', $_REQUEST) && array_key_exists('lang', $_REQUEST)) {
-            $this->handleExport(
-                sanitize_key($_REQUEST['type']),
-                sanitize_key($_REQUEST['lang'])
-            );
-        }
-        wp_redirect(remove_query_arg(['type', 'action', 'lang']));
+        parent::register();
+        $this->doRegisterScripts();
     }
 
-    protected function generateSkuFromTitleAction()
+    private function doRegisterScripts(): void
+    {
+        wp_enqueue_script('exportScript', plugin_dir_url(__FILE__).'../../static/backoffice.pages.tabs.export.js');
+        wp_enqueue_script('exportSweetalertScript', plugin_dir_url(__FILE__).'../../static/vendors/sweetalert2/sweetalert.min.js');
+        // So we can pass values to javascript file
+        wp_localize_script('exportScript', 'exportSettings',
+        [
+            'url' => rest_url(EndpointLoader::getFullNamespace().'/'.ExportEndpoint::ROUTE),
+            'translations' => [
+                'Your file has been generated' => __('Your file has been generated', I18N::DOMAIN),
+                'Your download will start in a few seconds. If not, you can download the file manually using the link below' => __('Your download will start in a few seconds. If not, you can download the file manually using the link below', I18N::DOMAIN),
+                'Please wait and keep the page and popup window open while we are preparing your export' => __('Please wait and keep the page and popup window open while we are preparing your export', I18N::DOMAIN),
+                'Preparing export' => __('Preparing export', I18N::DOMAIN),
+                'Stop exporting' => __('Stop exporting', I18N::DOMAIN),
+                'Size' => __('Size', I18N::DOMAIN),
+                'Export failed' => __('Export failed', I18N::DOMAIN),
+                'Something went wrong during export or server timed out. You can try manual export via command line, do you want to read the guide?' => __('Something went wrong during export or server timed out. You can try manual export via command line, do you want to read the guide?', I18N::DOMAIN),
+                'No, thanks' => __('No, thanks', I18N::DOMAIN),
+                'Yes, please' => __('Yes, please', I18N::DOMAIN),
```

Comment:
those translations should be escaped, cos you directly show the in html

Reply:
Used `esc_html__` for translations.